packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -1424,10 +1448,18 @@ Hardhat Network's forking functionality only works with blocks from at least spu
         if (txHash.equals(hash)) {
           const vmDebugTracer = new VMDebugTracer(vm);
           return vmDebugTracer.trace(async () => {
-            await vm.runTx({ tx: txWithCommon, block });
+            await vm.runTx({
+              tx: txWithCommon,
+              block,
+              skipHardForkValidation: true,
```

Comment:
This is a new check that we need to disable in the `runTx` calls. If this is not enabled, `runTx` will fail if the VM and the block have a different hardfork. To be honest, I'm not sure in exactly which scenarios this happens, but several tests passed again with this change :man_shrugging:

Reply:
I'll double check this

Reply:
Oh, this can trace remote transactions, so see https://github.com/NomicFoundation/hardhat/pull/3767/files#r1143994004